_posts/2020-05-07-removing-capybara-webkit.md

```diff
@@ -0,0 +1,133 @@
+
+And now we tell capybara to use one of those drivers for all the tests that require javascript:
+
+```ruby
+Capybara.javascript_driver = :headless_chrome # or :headless_firefox
+```
+
+> We had an extra step here because a lot of specs had the drivers specified on the actual test `describe` block like this: `describe "Do something", js: true, driver: :webkit do`. Instead of replacing the driver there, we just removed that option on all the tests so it uses the base config we set above.
+
+And lastly, before you run your test, make you you have no references to capybara-webkit in your code (search for `capybara-webkit`, `Capybara::Webkit` and `:webkit` strings).
+
+# Fixing Specs
+
+We first make sure tests are working locally so we are sure we don't have errors due to a misconfigured container.
+
+This application uses an old version of [Chosen](https://harvesthq.github.io/chosen/) to customize the `select` tags. The test suite includes a helper method so the driver can select options from that custom select and that helper method started failing.
+
+We were using a helper method from this [gist](https://gist.github.com/thijsc/1391107/699d65defed793eed0f04ead33c35737c641be53) that relied too much on `page.execute_script` and `page.evaluate_script`. The solution for this was to use a different method calling only capybara methods to actually mimic the user interactions.
+
+```diff
+  def select_from_chosen(item_text, options)
+-    field = find('#' + options[:from], visible: false)
+-    option_value = page.evaluate_script("$(\"##{field[:id]} option:contains('#{item_text}')\").val()")
+-    page.execute_script("value = ['#{option_value}']\; if ($('##{field[:id]}').val()) {$.merge(value, $('##{field[:id]}').val())}")
+-    option_value = page.evaluate_script("value")
+-    page.execute_script("$('##{field[:id]}').val(#{option_value})")
+-    page.execute_script("$('##{field[:id]}').trigger('liszt:updated').trigger('change')")
+
++    field = find_field(options[:from], visible: false)
++    find("##{field[:id]}_chzn").click
++    find("##{field[:id]}_chzn ul.chzn-results li", text: item_text).click
+  end
+```
+
+So we not only updated the gem, we also improved the test suite by not doing obscure Javascript calls.
+
+# Updating Docker Container
+
+Now that our tests are passing we can update the `Dockerfile` removing the commands that installed QT's libraries `qt5-default` and `libqt5webkit5-dev` (that will depend on the OS your are using, we have a Debian image so we just removed those packes from the `apt-get` command).
+
+Then we make sure we have Mozilla Firefox or Google Chrome installed. It was easier to install Firefox since the `firefox-esr` package was already available on the Debian Buster image we were using. You can use a Docker image which includes other browsers or install Google Chrome if you want to.
+
+So we changed the default driver to use Firefox:
+
+```ruby
+Capybara.javascript_driver = :headless_firefox
```

Comment:
if it was google chrome would you do `:headless_chrome` ?

Reply:
yes, the example using headless chrome is on the Replacing section after I have that notice to make sure the browsers are updated